Infrared: add remote adapter, add terminology (#46548)
home-assistant/home-assistant.github.io: source/_integrations/infrared.markdown

The **Infrared** {% term integration %} acts as an abstraction layer between IR hardware (such as ESPHome) and device-specific integrations that send commands to infrared-controlled devices, or receive signals from infrared remotes.

{% include integrations/building_block_integration.md %}

## Finding and adding integrations that support infrared

The **Infrared** {% term integration %} is a building block that other integrations build on. To control an IR-controlled device or react to an infrared remote, you need an integration that supports infrared, such as one for your specific TV or air conditioner.

To find and add integrations that support infrared:

1. In the Home Assistant documentation, go to the [Infrared category](/integrations/#infrared).
2. Browse the listed integrations to find one that matches your IR-controlled device.
3. Open the integration's page and follow its setup instructions.

> ## Controlling devices without a wired infrared emitter

Reviewer: this section is confusing. the title is about non-wired emitters, but then the first point is about wired ones.
can we instead link to the list of existing integrations that provide `infrared` entities, instead of mentioning specific ones here?

Reviewer 2: More generally, when searching for infrared integrations the discoverability of integrations that provide infrared and integrations that consume them, is all jumbled up. While there are only a few now its ok, but I guess this will scale up a lot as more integrations add support.

To control an IR-controlled device, an infrared emitter has to sit close to it, with line-of-sight to the device. There are two options to achieve this:

> - Using a wired IR emitter. A wired emitter, such as a USB-UIRT, plugs into your Home Assistant server, so it can only reach devices near the server, which is rarely in the same room as your TV or air conditioner.

Reviewer: I don't think we support USB-UIRT. did I miss it?

Author: Thank you for the catch! removed.

- Using an infrared remote adapter (proxy). An infrared remote adapter, such as the [Seeed Studio XIAO IR Mate](https://www.seeedstudio.com/XIAO-Smart-IR-Mate-p-6492.html), is a small ESPHome device with a built-in emitter that connects to Home Assistant over Wi-Fi (or Ethernet). You place it next to the IR-controlled device, and it sends IR commands over your network on behalf of Home Assistant, letting Home Assistant control IR-controlled devices anywhere in your home.

When you set up an ESPHome device configured as an infrared remote adapter, Home Assistant automatically discovers its built-in infrared emitters. Each emitter appears as an infrared entity, so you can select it when configuring an integration for an IR-controlled device.

### To set up an infrared remote adapter to control devices

> 1. Get an ESPHome remote adapter. You have two options:

Reviewer: Should this be in the ESPHome integration page instead?
ESPHome is not the only infrared entity provider integration.

Reviewer 2: To me, this makes (the entire page) sound like its documenting an ESPHome feature, and not a generic feature.

Author: Excellent point. Thank you!

3. Set up the infrared remote adapter through the [ESPHome](/integrations/esphome/) integration. Home Assistant discovers its built-in infrared emitters and surfaces them as infrared entities.
    - Depending on the device, you can follow the instructions in the [Ready-made Projects](https://esphome.io/projects/) section of the ESPHome documentation.
4. In Home Assistant, add the integration for your IR-controlled device, such as [LG Infrared](/integrations/lg_infrared/). During that integration's setup, when you are asked which infrared emitter to use, select the emitter from your infrared remote adapter. For more information, refer to [Finding and adding integrations that support infrared](#finding-and-adding-integrations-that-support-infrared).
5. If you need to control IR-controlled devices in different rooms, place multiple infrared remote adapters around your home and select the one closest to each device.

## About infrared terminology

This section explains some of the key terms on this page and how they are used in the Home Assistant documentation.

### Infrared remote adapter (proxy)

> A network-connected device, typically running ESPHome, that relays IR commands on behalf of Home Assistant over Wi-Fi or Ethernet. An infrared remote adapter is also known as an infrared proxy.

Reviewer: ```suggestion
A device that relays IR commands on behalf of Home Assistant. It can connect to Home Assistant over Wi-Fi, Ethernet, or other technology. An infrared remote adapter is also known as an infrared proxy.
```

zigbee ones will also show up eventually

Author: thx

It has the following characteristics:

- The infrared remote adapter contains one or more emitters, and sometimes a receiver.

> - A remote adapter needs to be placed near the IR-controlled device.

Reviewer: this is not necessarily true. the emitter can be placed away from the device.
a receiver can also be placed anywhere if it is used to receive without being tied to the device

- The infrared remote adapter is the hardware device. The emitters and receivers it contains are the parts that each become an infrared entity.

#### Infrared emitter

The component that sends out IR signals, the same kind of signal a physical remote sends. An emitter is what actually controls a device, and it is part of a piece of hardware, such as an infrared remote adapter. In Home Assistant, each emitter appears as an infrared entity that other integrations can use.

#### Infrared receiver

> The component that catches IR signals sent by a device like a handheld remote. Home Assistant can use those signals, for example to learn remote codes or trigger automations. Like an emitter, a receiver is part of a piece of hardware, such as an infrared remote adapter. Each receiver appears as an infrared entity.

Reviewer: ```suggestion
The component that catches IR signals sent by a device like a handheld remote. Home Assistant can use those signals, for example to update entities state or trigger automations. Like an emitter, a receiver is part of a piece of hardware, such as an infrared remote adapter. Each receiver appears as an infrared entity.
```

  - '@home-assistant/core'
ha_integration_type: entity
related:
  - docs: /integrations/#infrared
    title: Integrations supporting infrared
  - url: https://esphome.io/projects/?type=irrf
    title: ESPHome infrared and radio frequency proxy projects
  - url: https://esphome.io/components/ir_rf_proxy/
    title: ESPHome infrared and radio frequency proxy component
---

> The **Infrared** {% term integration %} acts as an abstraction layer between an infrared remote adapter (proxy) and device-specific integrations (such as [LG Infrared](/integrations/lg_infrared/), [Samsung Infrared](/integrations/samsung_infrared/), or [SMLIGHT SLZB](/integrations/smlight/)) to send commands to infrared-controlled devices, or to receive signals from infrared remotes.

Reviewer: smlight is a "provider" integration (creates `infrared` entities), while lg_infrared and samsung_infrared are "consumer" integrations (uses `infrared` entities).
This sentence mixes both. We should probably remove smlight from this sentence, and add another paragraph saying something like "You need an integration that provides infrared entities by integrating a infrared remote adapter, such as esphome or smlight ".

How the Infrared integration connects device integrations to the emitters and receivers of an infrared remote adapter.
</p>

{% include integrations/building_block_integration.md %}

## Finding and adding integrations that support infrared

The **Infrared** {% term integration %} is a building block that other integrations build on. To control an infrared-controlled device or react to an infrared remote, you typically need both infrared hardware that exposes an infrared emitter or receiver entity and an integration for your specific device.

To set up infrared hardware and a device integration:

1. Add an integration that provides infrared entities, such as [Broadlink](/integrations/broadlink/), [SMLIGHT SLZB](/integrations/smlight/), or an ESPHome infrared proxy.
2. Add the integration for your infrared-controlled device (for example, [LG Infrared](/integrations/lg_infrared/)) and select the infrared emitter or receiver entity when prompted.

> ## Controlling devices from Home Assistant without a wired infrared emitter

Reviewer: is there a reason to mention wired emitters at all?

Reviewer: after reading the body of this section, I would suggest removing it completely. the other sections already cover the need for a device

Author: good point! Removed it.

> If you do not want to use a wired infrared emitter plugged into to Home Assistant, you can use an infrared remote adapter (proxy). This is a small device that can connects to Home Assistant not only over Ethernet and other methods, but also over Wi-Fi. You place it within line-of-sight of the infrared-controlled device, and it sends IR commands over your network on behalf of Home Assistant.

Reviewer: "not only over Ethernet and other methods, but also over Wi-Fi." 

this seems to make Wi-Fi something special, and the sentence reads a bit weird. maybe just "over Ethernet, Wi-Fi, or other methods"? Although I am not even sure we need to state this at all. 
A device can connect over anything, there are no restrictions from the `infrared` integration PoV. I suggest removing this sentence.

### Prerequisites

- An infrared remote adapter. If you're unsure what to get:

>   - Find integrations that support infrared: In the documentation, search for the [infrared category](/integrations/#infrared).

Reviewer: this link shows both integrations that provide infrared entities and integrations that consume them. 
maybe we should not set the `infrared` category on the consumers, to make it easier to find them? it also aligns well with how we usually do it for other entity domains (i.e., we only add "sensor" to an integration that creates sensor entities)

Author: but then how do you find the consumers?
